code/src/sixsq/nuvla/server/resources/api_docs/openapi.clj

```diff
@@ -0,0 +1,122 @@
+                  403 {:description "Not authorized"}
+                  404 {:description "The resource was not found"}}}})
+
+
+(defn collection-query
+  "Returns the OpenAPI method spec for the query operation"
+  [{resource-type-name :name :as _resource-metadata}]
+  {:get
+   {:tags        [resource-type-name]
+    :description (str "Returns a collection of " resource-type-name)
+    :responses   {}}})
+
+
+(defn resource-paths
+  "Returns the `paths` section of the OpenAPI spec for the specified resource."
+  [{:keys [type-uri scrud-operations] resource-type-name :name :as resource-metadata}]
+  (let [scrud-op-supported?  (set scrud-operations)
+        resource-param "resource-id"
+        resource-ops   (merge
+                         (when (scrud-op-supported? :get)
+                           (resource-get resource-metadata)))
+        collection-ops (merge
+                         (when (scrud-op-supported? :query)
+                           (collection-query resource-metadata)))]
+    (merge
+      (when (seq collection-ops)
+        {(str "/" type-uri) collection-ops})
+      (when (seq resource-ops)
+        {(str "/" type-uri "/{" resource-param "}")
+         (merge
+           {:parameters [{:name        resource-param
+                          :in          "path"
+                          :required    true
+                          :description (str "The ID of the " resource-type-name)
+                          :schema      {:type   "string"
+                                        :format "uuid"}}]}
+           resource-ops)}))))
+
+
+(defn paths
+  "Returns the `paths` section of the OpenAPI spec."
+  [resources-metadata]
+  (reduce (fn [m resource-metadata]
+            (merge m (resource-paths resource-metadata)))
+          (sorted-map)
+          resources-metadata))
+
+
+(defn schemas
+  "Returns the `schemas` section of the OpenAPI spec."
+  [resources-metadata]
+  (->> resources-metadata
+       (map openapi-schema/->openapi-schemas)
+       (apply merge)
+       (into (sorted-map))))
+
+
+(defn openapi-spec
+  "Returns the OpenAPI spec for the given resource types."
+  [resources-metadata]
+  (let [nuvla-api-spec {:openapi      "3.0.3"
+                        :info         {:title          "Nuvla API"
+                                       :description    "HTTP-based RESTful API, for the management of Nuvla resources."
+                                       :termsOfService "https://docs.nuvla.io/legal/legal/"
+                                       :contact        {:name  "API Support",
+                                                        :email "apiteam@nuvla.io"}
+                                       :license        {:name "Apache 2.0",
+                                                        :url  "https://www.apache.org/licenses/LICENSE-2.0.html"}
+                                       ;; TODO: add the api version
+                                       :version        "<version>"}
```

Comment:
Where can I pull the api version from?

Reply:
I think we don't have yet an utility function for that.

Some ways to get the api version:
Maybe from the api server jar manifest. Maybe we can generate a file with version into [resources folder](https://github.com/nuvla/api-server/blob/d98be1fe01d8d763098d9128e9a800cae401e1b9/code/resources).

But it's not an easy question to decide which version to show. Because there is the case where the api-server resources are extended with a private part from another clojure project that add some extra resources.